test-app/tests/integration/components/file-input-field-test.gts

```diff
@@ -0,0 +1,274 @@
+    await render(<template>
+      <FileInputField
+        @label="Label"
+        @error="Error text"
+        data-file-input-field
+      />
+    </template>);
+
+    let error = find('[data-error]');
+
+    assert.dom(error).hasText('Error text');
+    assert.dom(error).hasAttribute('id');
+
+    let errorId = error?.getAttribute('id') || '';
+    assert.ok(errorId, 'Expected errorId to be truthy');
+
+    // For the file input field component, the only aria-describedby
+    // value should be the errorId.  This is due to the way the component
+    // is structured, where the label+hint are rendered inside of the
+    // wrapping <label> element
+    let describedby =
+      find('[data-file-input-field]')?.getAttribute('aria-describedby') || '';
+    assert.ok(
+      describedby.includes(errorId),
+      'Expected errorId to be included in the aria-describedby'
+    );
+
+    assert.dom('[data-file-input-field]').hasAttribute('aria-invalid', 'true');
+
+    assert.dom('[data-file-input-field]').hasClass('shadow-error-outline');
+    assert
+      .dom('[data-file-input-field]')
+      .doesNotHaveClass('shadow-focusable-outline');
+  });
+
+
+  test('it sets the "for" attribute on the label to the "id" attribute of the file input field', async function (assert) {
+    await render(<template>
+      <FileInputField @label="Label" data-file-input-field />
+    </template>);
+
+    let labelFor = find('label')?.getAttribute('for') || '';
+    assert.ok(labelFor, 'Expected the id attribute of the label to be truthy');
+
+    assert
+      .dom('[data-file-input-field]')
+      .hasAttribute(
+        'id',
+        labelFor,
+        'Expected the for attribute on the label to match the id attribute on the checkbox'
+      );
+  });
+
+  test('it disables the file input using @isDisabled', async function (assert) {
+    await render(<template>
+      <FileInputField
+        @label="Label"
+        @isDisabled={{true}}
+        data-file-input-field
+      />
+    </template>);
+
+    assert.dom('[data-file-input-field]').isDisabled();
+    assert.dom('[data-file-input-field]').hasClass('text-disabled');
+  });
+ 
+  test('it spreads attributes to the underlying input', async function (assert) {
+    await render(<template>
+      <FileInputField
+        @label="Label"
+        placeholder="Placeholder text"
+        data-file-input-field
+      />
+    </template>);
+
+    assert
+      .dom('[data-file-input-field]')
+      .hasAttribute('placeholder', 'Placeholder text');
+  });
+
+  test('it can upload a file and display the filename', async function (assert) {
+    await render(<template>
+      <FileInputField @label="Label" data-file-input-field />
+    </template>);
+
+    const file = createFile(['Upload file sample'], {
+      name: 'sample.txt',
+      type: 'text/plain',
+    });
+
+    triggerEvent('[data-file-input-field]', 'change', { files: [file] });
+
+    await settled();
```

Comment:
I think this could be simplified to

```suggestion
    await triggerEvent('[data-file-input-field]', 'change', { files: [file] });
```

Or didn't this work?

Reply:
I didn't know `triggerEvent` was async...TIL.

Reply:
Yeah, pretty much all helpers from `@ember/test-helpers` are. You can think of them as having `await settled()` as their last statement internally, so you should never need another `settled()` call directly after those!